src/components/organisms/Badges/Badges.tsx

```diff
@@ -48,13 +39,36 @@ export const Badges: React.FC<{
           >)
       ) ?? [];
 
+    const venuesRequests = visits.map((visit) =>
+      firestore.collection("venues").where("name", "==", visit.id).get()
+    );
+
+    const venues: WithId<AnyVenue>[] = [];
+    const hasVenuesRequests = isTruthy(venuesRequests);
+
+    // If there are no venues visited avoid sending the request.
+    if (hasVenuesRequests) {
+      const requestSnapshots = await Promise.all(venuesRequests);
+
+      // Promise all returns arrays as response. That's why there is so much depth.
+      // TODO: Same logic can be used for the private chats as well.
+      requestSnapshots.forEach((venuesSnapshot) =>
+        venuesSnapshot.docs.forEach((venueSnapshot) =>
```

Comment:
you could potentially `.flatMap` this? Which should then also allow you to avoid using array mutation with the `.push[]` as you can just return the correct values as a new array from the `.flatMap`